src/promptflow/promptflow/executor/_service/utils/process_utils.py

```diff
@@ -0,0 +1,95 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import contextlib
+import json
+import multiprocessing
+import os
+from datetime import datetime, timedelta
+from typing import Callable
+
+from promptflow._core._errors import UnexpectedError
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.exception_utils import ExceptionPresenter, JsonSerializedPromptflowException
+from promptflow._utils.logger_utils import service_logger
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._service._errors import ExecutionTimeoutError
+from promptflow.executor._service.utils.service_utils import get_log_context
+
+LONG_WAIT_TIMEOUT = timedelta(days=1).total_seconds()
+SHORT_WAIT_TIMEOUT = 10
+
+
+async def invoke_sync_function_in_process(
+    request, context_dict: dict, target_function: Callable, wait_timeout: int = LONG_WAIT_TIMEOUT
```

Comment:
I would prefer the interface like this:
```invoke_sync_function_in_process(func, args, kwargs)```
instead of too specific.

Reply:
All errors/result passing logic would be better hide in the function.

Reply:
Updated.